lib/pbench/test/unit/server/query_apis/test_index_mappings.py

```diff
@@ -0,0 +1,62 @@
+import pytest
+from http import HTTPStatus
+from sqlalchemy.exc import DatabaseError
+
+from pbench.server.database.models.template import Template
+
+
+@pytest.fixture()
+def database_error(monkeypatch):
+    def raise_db_error(name: str):
+        raise DatabaseError("DB Error")
+
+    with monkeypatch.context() as m:
+        m.setattr(Template, "find", raise_db_error)
+        yield
+
+
+class TestIndexMappings:
+    """
+    Unit testing for resources/IndexMappings class.
+
+    In a web service context, we access class functions mostly via the
+    Flask test client rather than trying to directly invoke the class
+    constructor and `post` service.
+    """
+
+    def test_query(self, client, server_config, find_template):
+        """
+        Check the construction of index mappings API and filtering of the
+        response body.
+        """
+        with client:
+            response = client.get(f"{server_config.rest_uri}/index/mappings")
+            assert response.status_code == HTTPStatus.OK
+            res_json = response.json
+            assert isinstance(res_json, dict)
+            assert len(res_json.keys()) == 5
+            assert list(res_json.keys()) == [
+                "@metadata",
+                "authorization",
+                "host_tools_info",
+                "run",
+                "sosreports",
+            ]
```

Comment:
I'm not sure that there is any point to the assertion at line 36 -- I think it's safe to assume that `response.json` will yield a `dict`.  And, if you're going to explicitly check the keys returned, I don't think there is any point to checking the count of keys.

As for the third assertion, given that we know what the input mapping was, we should be able to predict exactly what the output will be, so I think we should be able to compare `res_json` directly to the full return value (which, presumably we would import from `conftest.py` somehow, either as a fixture or just a constant).  (If we _can't_ do that, then I think maybe there's a problem with this interface.)